packages/notification-center/src/components/notification-center/components/user-preference/ChannelPreference.tsx

```diff
@@ -1,17 +1,45 @@
 import React from 'react';
-import { Text, TextBlock } from './UserPreference';
 import { useNovuThemeProvider } from '../../../../hooks/use-novu-theme-provider.hook';
 import { Switch } from '@mantine/core';
 import { getChannel } from './channels';
 import styled from 'styled-components';
+import { useApi } from '../../../../hooks';
+import { switchStyles, Text, TextBlock } from './styles';
+import { IUserPreferenceSettings } from '../../../../index';
 
-export function ChannelPreference({ type, active }) {
+interface IChannelPreferenceProps {
+  type: string;
+  active: boolean;
+  templateId: string;
+  index: number;
+  setSettings: (IUserPreferenceSettings) => void;
+}
+export function ChannelPreference({ type, active, templateId, index, setSettings }: IChannelPreferenceProps) {
   const { label, description, Icon } = getChannel(type);
   const { theme } = useNovuThemeProvider();
+  const { api } = useApi();
   const baseTheme = theme?.notificationItem?.unseen;
   const primaryColor = baseTheme.fontColor;
   const secondaryColor = baseTheme.timeMarkFontColor;
 
+  const handleUpdateChannelPreference = async (checked) => {
+    const result = await api.updateSubscriberPreference(templateId, type, checked);
```

Comment:
Maybe we should add here a loading state, add a loader to the actual clicked switch, and disable all the others.